src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,104 @@
+        try {
+            Files.createDirectories(Paths.get("data/"));
+            File file = new File("data/duke.txt");
+
+            //Check if the file exists
+            if (!file.exists()) {
+                boolean r = file.createNewFile();
+            }
+            return new Storage(file);
+        }
+        catch(IOException e){
+            throw new DukeException("☹ OOPS!!! Error occurs when try to create a new data file");
+        }
+    }
+
+    /**
+     * Save the taskList to the file
+     * @param taskList a list of tasks
+     * @throws DukeException Exceptions that are possible to occur in Duke
+     */
+    public void saveToFile(TaskList taskList) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(file, false);
+            for (int i=0; i<taskList.size(); i++) {
+                Task task= taskList.get(i);
+                fw.write(task.saveTask());
+                fw.write("\r\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException("☹ OOPS!!! Error occurs when save the data");
+        }
+    }
+
+    /**
+     * Read the file and reload the data from the file
+     * @param taskList a list of tasks
+     * @throws DukeException Exceptions that are possible to occur in Duke
+     */
+    public void readFile(TaskList taskList) throws DukeException {
+        try {
+            Scanner s = new Scanner(file);
+            String current, desc, name;
+            LocalDate time;
+            char type;
+            while(s.hasNext()) {
+                current = s.nextLine();
+                type = current.charAt(1);
+                if (type == 'T') {
+                    Task newTask = new Todo(current.substring(7));
+                    newTask.setCompleted(current.charAt(4)=='X');
+                    taskList.add(newTask);
+                }
+                else if (type == 'E') {
+                    desc = current.substring(7);
+                    String[] split = desc.split("->at: ", 2);
+                    name = split[0];
+                    time = LocalDate.parse(split[1]);
+                    Task newTask = new Event(name, time);
+                    newTask.setCompleted(current.charAt(4)=='X');
+                    taskList.add(newTask);
+                }
+                else {
+                    desc = current.substring(7);
+                    String[] split = desc.split("->by: ", 2);
+                    name = split[0];
+                    time = LocalDate.parse(split[1]);
+                    Task newTask = new Deadline(name, time);
+                    newTask.setCompleted(current.charAt(4)=='X');
+                    taskList.add(newTask);
+                }
+            }
+        } catch (IOException e) {
+            throw new DukeException("OOPS!!! Error occurs when reload the data");
+        }
+    }
```

Comment:
This method is too long. Would it be better to move each `if` / `else-if` block to its respective helper methods?